mne_bids/read.py

```diff
@@ -257,10 +257,7 @@ def _handle_channels_reading(channels_fname, bids_fname, raw):
         bad_bool = [True if chn.lower() == 'bad' else False
                     for chn in channels_dict['status']]
         bads = np.asarray(channels_dict['name'])[bad_bool]
-
-        # merge with bads already present in raw data file (if there are any)
-        unique_bads = set(raw.info['bads']).union(set(bads))
-        raw.info['bads'] = list(unique_bads)
+        raw.info['bads'] = bads
```

Comment:
the problem is that this column is optional, so I think we should still default to `raw.info['bads']` if it doesn't exist

Reply:
I disagree. The specs say:

> OPTIONAL. Data quality observed on the channel (good/bad). A channel is considered bad if its data quality is compromised by excessive noise. Description of noise type SHOULD be provided in [status_description].

If users wish to mark channels as bad, they should do so via this file. Otherwise – where do you stop? Would you also start merging events from the raw data and sidecar files? Isn't the reasoning behind the human-readable sidecars that one can get an idea about the data without having to read the raw files?

Reply:
we do the same for sampling frequency. But there is a priority order of course.

does raw data have events? I think not.

Reply:
> does raw data have events? I think not.

BIDS stores them as annotations.

Reply:
right but you can't get them from `fif` files, so no ambiguity?

Reply:
You can. I mean you can read and modify the Annotations in the fif and believe this is The Way To Go (TM) because if you read data via MNE-BIDS and want to get the events, you call `events_from_annotations` on the Raw. This could lead you to beleive that to change the events, you have to edit the Annotations -- while in fact you should modify `events.tsv`.

Reply:
Keep in mind that we're only talking about data that is already in BIDS format here!

Reply:
Maybe that's where our premises are different

Reply:
humm ... I didn't realize you could save `annotations` in fif files. I think that would only be readable in MNE-Python though and not in MNE-C and Elekta software? anyway, that's a bunny trail

so what's the workflow in your mind to edit annotations? Does it involve modifying the events.tsv file directly? Or to read it in python using MNE-BIDS, edit it and write it back? I now realize that editing sidecar text files are not so straightforward as it seems. Because you need to check consistency, for e.g., if the annotation duration doesn't exceed the total recording duration.

Reply:
> so what's the workflow in your mind to edit annotations? 

BIDS doesn't know annotations, only events. The use of Annotations by MNE-BIDS is just an implementation detail. MNE-BIDS populates the Annotations based on `events.tsv`; and users then retrieve the events via `mne.events_from_annotations()`. An alternative, obviously, could be the introduce a function `mne_bids.read_events()` (which might take an optional parameter to return an Annotations object instead an MNE event array).

> Does it involve modifying the events.tsv file directly?

If you ask me, noone should ever touch the sidecar files manually; but apparently this is what BIDS is intended for, so… yeah, I guess the process would involve modifying events.tsv manually.

> Or to read it in python using MNE-BIDS, edit it and write it back?

That would be my preferred way of doing it.

> I now realize that editing sidecar text files are not so straightforward as it seems. Because you need to check consistency, for e.g., if the annotation duration doesn't exceed the total recording duration.

That's why one should run the validator after every edit, and the validator should check those things.

Reply:
Oh and maybe I should add that `read_raw_bids()` entirely replaces `Raw.annotations` with the Annotations populated from `events.tsv` -- there's no merge taking place with existing Annotations in the raw file.